sdk/servicebus/Azure.Messaging.ServiceBus/src/Processor/ProcessMessageEventArgs.cs

```diff
@@ -117,5 +153,107 @@ await _receiver.AbandonMessageAsync(message, propertiesToModify, cancellationTok
+            return TrackMessagesAsReceived(messages);
+        }
+
+        /// <summary>
+        /// Receives a list of deferred <see cref="ServiceBusReceivedMessage"/> identified by <paramref name="sequenceNumbers"/>.
+        /// Messages received using this method are subject to the behavior defined in the <see cref="ServiceBusProcessorOptions.AutoCompleteMessages"/>
+        /// and <see cref="ServiceBusProcessorOptions.MaxAutoLockRenewalDuration"/> properties.
+        /// </summary>
+        ///
+        /// <param name="cancellationToken">An optional <see cref="CancellationToken"/> instance to signal the request to cancel the operation.</param>
+        /// <param name="sequenceNumbers">An <see cref="IEnumerable{T}"/> containing the sequence numbers to receive.</param>
+        ///
+        /// <returns>Messages identified by sequence number are returned.
+        /// Throws if the messages have not been deferred.</returns>
+        /// <seealso cref="DeferMessageAsync(ServiceBusReceivedMessage, IDictionary{string, object}, CancellationToken)"/>
+        /// <exception cref="ServiceBusException">
+        ///   The specified sequence number does not correspond to a message that has been deferred.
+        ///   The <see cref="ServiceBusException.Reason" /> will be set to <see cref="ServiceBusFailureReason.MessageNotFound"/> in this case.
+        /// </exception>
+        public virtual async Task<IReadOnlyList<ServiceBusReceivedMessage>> ReceiveDeferredMessagesAsync(
+            IEnumerable<long> sequenceNumbers,
+            CancellationToken cancellationToken = default)
+        {
+            ValidateCallbackInScope();
+            IReadOnlyList<ServiceBusReceivedMessage> messages = await _receiver.ReceiveDeferredMessagesAsync(sequenceNumbers, cancellationToken).ConfigureAwait(false);
+            return TrackMessagesAsReceived(messages);
+        }
+
+        private IReadOnlyList<ServiceBusReceivedMessage> TrackMessagesAsReceived(IReadOnlyList<ServiceBusReceivedMessage> messages)
+        {
+            // manager would be null in scenarios where customers are using the public constructor for testing purposes.
+            if (_manager?.ShouldAutoRenewMessageLock() == true)
+            {
+                foreach (ServiceBusReceivedMessage message in messages)
+                {
+                    Messages[message] = default;
+                    _renewalTasks[_manager.RenewMessageLockAsync(message, _lockRenewalCancellationSource)] = default;
+                }
+            }
+            else
+            {
+                foreach (ServiceBusReceivedMessage message in messages)
+                {
+                    Messages[message] = default;
+                }
+            }
+
+            return messages;
+        }
+
+        internal void EndExecutionScope()
```

Comment:
To be honest, I wish we had implemented this for all methods in the args to make the contract more explicit. It wasn't strictly necessary to add here, but I think it can prevent users from shooting themselves in the foot.